src/EventSubscriber/CartEventSubscriber.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\commerce_option\EventSubscriber;
+
+use Drupal\commerce_cart\Event\CartEvents;
+use Drupal\commerce_price\Calculator;
+
+use Drupal\Core\TempStore\PrivateTempStoreFactory;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Class CartEventSubscriber.
+ */
+class CartEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Drupal\Core\TempStore\PrivateTempStoreFactory definition.
+   *
+   * @var \Drupal\Core\TempStore\PrivateTempStoreFactory
+   */
+  protected $tempstorePrivate;
+
+  /**
+   * Constructs a new CartEventSubscriber object.
+   */
+  public function __construct(PrivateTempStoreFactory $tempstore_private) {
+    $this->tempstorePrivate = $tempstore_private;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events[CartEvents::CART_ENTITY_ADD][] = ['addOptions'];
+    $events[CartEvents::CART_ENTITY_ADD][] = ['mergeOrderItems'];
+
+    return $events;
+  }
+
+  /**
+   * Adds the option values to the new order item.
+   *
+   * @param \Symfony\Component\EventDispatcher\Event $event
+   *   The dispatched event.
+   */
+  public function addOptions(Event $event) {
+    $variation = $this->tempstorePrivate->get('commerce_option')->get('variation_id');
```

Comment:
Could you please rename this to $variation_id as it's really the variation ID we're fetching and avoids any confusion.